### PR TITLE
Migrate ink! docs, Write Contract module docs

### PR DESCRIPTION
**Pull request by shawntabrizi**
The goal here is to set a base level set of documentation which introduces the smart contract ecosystem of substrate.

Ping @robbepop and @pepyakin

### CODE REVIEW

File: docs/contracts/introduction.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+* AccountId
+* Balance
+* Associated Code
+* Storage
+
+The [halting problem](https://en.wikipedia.org/wiki/Halting_problem) is solved by using a gas/fee system.
+
+The Contract module provides safety to the blockchain state by enabling revertable transactions, which rollback any changes to the storage by contract calls which fail.
+
+Contract calls can change alter the storage of the contract, create new contracts, and call other contracts.
```
Comment (pepyakin) [inline docs/contracts/introduction.md]: Either

```suggestion
Contract calls can alter the storage of the contract, create new contracts, and call other contracts.
```
or
```suggestion
Contract calls can change the storage of the contract, create new contracts, and call other contracts.
```

File: docs/contracts/introduction.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+### Differences to Ethereum
+
+Unlike Ethereum, the Contracts module uses a WebAssembly interpreter ([wasmi](https://github.com/paritytech/wasmi/)) to execute smart contracts.
```
Comment (pepyakin) [inline docs/contracts/introduction.md]: The interpreter is just an implementation detail. The important difference that Ethereum uses EVM and srml-contract uses WebAssembly as the VM.

File: docs/contracts/introduction.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+## ink!
+
+ink! is Rust based embedded domain specific language ([eDSL](https://wiki.haskell.org/Embedded_domain_specific_language)) for writing Wasm smart contracts specifically for the SRML Contract module. The main goals of ink! are user friendliness, conciseness, and efficiency.
+
+### Abstraction Layers
+
+The ink! language is composed of three different layers of abstractions with which you can write smart contracts:
+
+* [Lang](https://github.com/paritytech/ink/tree/master/lang): The actual eDSL to provide a user friendly interface to writing smart contract code.
+* [Model](https://paritytech.github.io/ink/ink_model/index.html): Medium-level abstractions to write smart contracts heavily inspired by [Fleetwood](https://github.com/paritytech/fleetwood).
```
[the remark this answers is not in the document]
Comment (shawntabrizi) [reply docs/contracts/introduction.md]: I have no idea what else to write for the model layer then... @robbepop?

File: docs/contracts/introduction.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+* [Core](https://paritytech.github.io/ink/ink_core/index.html): The core utilities and APIs used to interact with the Contract module.
+
+### Similarities to Solidity
+
+ink! should feel familiar in structure to developers who have used Solidity:
+
+* The skeleton of a contract has all of the same components that you might expect:
```
[the remark this answers is not in the document]
Comment (shawntabrizi) [reply docs/contracts/introduction.md]: This content has changed a bit, but if you can find a good place to plug it or a sentence or two you want to add, please do suggest and edit :)

File: docs/contracts/introduction.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+* ink! exposes a number of environment variables to the contract developer like the `caller`, `balance`, `gas_left`, and more!
+
+### Differences to Solidity
+
+Being based in Rust, the list of differences to Solidity far surpass what can be written here, but we will touch on some notable ones:
+
+* Through Rust, ink! can provide compile time overflow/underflow safety. Using a Rust compiler configuration, you can specify whether you want to support overflowing math, or if you want contract execution to panic when overflows occur. No need to continually import "Safe Math" libraries, although Rust also provides [integrated checked, wrapped, and saturated math functions](https://doc.rust-lang.org/std/primitive.u32.html).
+
+* Rust also provides a safe API for describing values which are not set with the [Option](https://doc.rust-lang.org/std/option/index.html) type. In Ethereum, you would use the `0x0` address to represent "no one" when checking ownership of an asset or burning tokens. With ink! and Rust, you can explicitly specify a value as `None` which has no disambiguous meaning.
```
[the remark this answers is not in the document]
Comment (shawntabrizi) [reply docs/contracts/introduction.md]: This has been updated as less of a comparison to solidity, and more of a hurrah to rust for getting rid of null.

Let me know what you think

File: docs/contracts/deploying-a-contract.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+You should start to see blocks being produced by your node in your terminal.
+
+You can interact with your node using the Polkadot UI:
+
+https://polkadot.js.org/apps/
+
+> **Note:** You will need to use Google Chrome to have this site interact with your local node. The Polkadot UI is hosted on a secure server, and your local node is not, which may cause compatibility issues on Firefox or Linux based Chromium. The other option is to [clone and run the Polkadot UI locally](https://github.com/polkadot-js/apps).
+
+If you go into the **Explorer** tab of the UI, you should also see blocks being produced!
```
Comment (kaichaosun) [inline docs/contracts/deploying-a-contract.md]: We need to set the node to be local before interacting with other parts.

File: docs/contracts/deploying-a-contract.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+![An image of the Substrate UI](./images/start-substrate-ui.png)
+
+## Creating the Flipper "Hello World" Contract
+
+This guide will not teach you about developing smart contracts on Substrate. For that, you can look at our [Substrate Smart Contract Development Tutorial](Writing-Your-First-Contract).
```
Comment (kaichaosun) [inline docs/contracts/deploying-a-contract.md]: `Writing-Your-First-Contract` is not a link. If it's not ready, maybe add a TODO tag.

File: website/i18n/en.json [hunk 8]
```diff
@@ -5,6 +5,24 @@
     "previous": "Previous",
     "tagline": "The place for blockchain innovators.",
     "docs": {
+      "contracts/cargo-contract": {
```
Comment (kaichaosun) [inline website/i18n/en.json]: We don't need en.json anymore.

File: docs/contracts/deploying-a-contract.md (hunk 6 above)
Comment (shawntabrizi) [reply docs/contracts/deploying-a-contract.md]: What do you mean by this?

File: docs/contracts/deploying-a-contract.md (hunk 7 above)
Comment (shawntabrizi) [reply docs/contracts/deploying-a-contract.md]: Good catch, this is an artifact from Readme.io and should be updated

File: docs/contracts/deploying-a-contract.md (hunk 6 above)
Comment (kaichaosun) [reply docs/contracts/deploying-a-contract.md]: In the `https://substrate-ui.parity.io/#/settings`, set  `remote node/endpoint to connect to` be local node.

**Pull request merged**